### ISSUE TITLE
Restraint doesn't wait for system to get IP address

### ISSUE THREAD

**Issue by jvillal-amp** [beaker-project/restraint#101]
Running restraint on a system and see in the console the following:

```
[    9.879394] restraintd[803]: * Fetching recipe: http://example.com:8000//recipes/207700/

Debian GNU/Linux 10 system1 ttyAMA0

sut03sys-r098-j3011cdm login: [   20.000607] restraintd[803]: * Fetching recipe: http://example.com:8000//recipes/207700/
[   30.000594] restraintd[803]: * Fetching recipe: http://example.com:8000//recipes/207700/
[   40.000462] restraintd[803]: * Fetching recipe: http://example.com:8000//recipes/207700/
[   40.000632] restraintd[803]: * WARNING **:While fetching recipe XML: Error resolving “example.com”: Temporary failure in name resolution
```

Restraint gives up trying to fetch the recipe..  I'm not sure why Restraint gives up though. It has nothing else to do. Seems like it should keep trying forever.

Note: changed real system name to 'example.com' in output

**Comment by jvillal-amp** [comment]
I have done a pull request on a proposed fix. With the patch it does the following:

```
[   14.598634] check_beaker[768]: /usr/bin/check_beaker: Waiting for IP address. Attempt: 2 of 12 failed

Debian GNU/Linux 10 sut03sys-r098-j3011cdm ttyAMA0

sut03sys-r098-j3011cdm login: [   19.602111] check_beaker[768]: /usr/bin/check_beaker: Waiting for IP address. Attempt: 3 of 12 failed
[   24.605349] check_beaker[768]: /usr/bin/check_beaker: Waiting for IP address. Attempt: 4 of 12 failed
[   29.608243] check_beaker[768]: /usr/bin/check_beaker: Waiting for IP address. Attempt: 5 of 12 failed
[   34.611082] check_beaker[768]: /usr/bin/check_beaker: Waiting for IP address. Attempt: 6 of 12 failed
[   39.614112] check_beaker[768]: /usr/bin/check_beaker: Waiting for IP address. Attempt: 7 of 12 failed
[   44.617094] check_beaker[768]: /usr/bin/check_beaker: Waiting for IP address. Attempt: 8 of 12 failed
[   49.619807] check_beaker[768]: /usr/bin/check_beaker: IP address acquired
[   49.645125] restraintd[829]: * Fetching recipe: http://bkr01sys-r098.scc-lab.amperecomputing.com:8000//recipes/207716/
[   49.723794] restraintd[829]: * Parsing recipe
[   49.724609] restraintd[829]: * Running recipe
[   49.724760] restraintd[829]: ** Fetching task: 396855 [/mnt/tests/dev.amp-dev.com/beaker/tasks/distribution-checkinstall.tgz]
[   50.290536] restraintd[829]: ** Preparing metadata
[   50.291937] restraintd[829]: ** Refreshing peer role hostnames: Retries 0
[   50.357896] restraintd[829]: ** Updating env vars
[   50.358689] restraintd[829]: ** Updating external watchdog: 2400 seconds
[   55.353185] restraintd[829]: ** Installing dependencies
[   55.356242] restraintd[829]: ** Running task: 396855 [/distribution/check-install]
[   57.335320] restraintd[829]: ** Completed Task : 396855
[   62.596394] restraintd[829]: * Finished recipe
```

**Comment by danrodrig** [comment]
@jvillal-amp as discussed in #102, we don't think that this issues is to be fixed in the `check_beaker` script. As you pointed out, restraint should keep retrying for fetching the recipe, so this should be fixed by increasing retries in restraint itself.
We are not 100% sure that an infinite loop would be safe, so increasing the restries to wait at least as much as what you found should be good.
We should drop #102, sorry for not letting you know before. If you want to open PR for increasing the retries in restraint, let us know, otherwise I can take care of it.

Thanks for the report and also for the PR.

**Comment by jvillal-amp** [comment]
@danrodrig  Thanks for the update! Uh, if you can do the PR that would be greatly appreciated as I admit at the moment I don't know what to update in the restraint code to do what you suggest.

**Comment by danrodrig** [comment]
> @danrodrig Thanks for the update! Uh, if you can do the PR that would be greatly appreciated as I admit at the moment I don't know what to update in the restraint code to do what you suggest.

@jvillal-amp no problem, I will take it. 🙂

**Comment by danrodrig** [comment]
@jvillal-amp this should be fixed with #110. Retries will be attempted for 120 seconds, instead of 30.

**Comment by jvillal-amp** [comment]
> @jvillal-amp this should be fixed with #110. Retries will be attempted for 120 seconds, instead of 30.

Thanks @danrodrig ! 👍
